packages/packages/v1-adapters/src/listeners/listeners.ts

```diff
@@ -0,0 +1,79 @@
+import { normalizeEvent } from './utils';
+import { CommandEventDescriptor, EventDescriptor, ListenerCallback, WindowEventDescriptor } from './types';
+
+const callbacksByEvent = new Map<EventDescriptor['name'], ListenerCallback[]>();
+let abortController = new AbortController();
+
+export function listenTo(
+	eventDescriptors : EventDescriptor | EventDescriptor[],
+	callback : ListenerCallback
+) {
+	if ( ! Array.isArray( eventDescriptors ) ) {
+		eventDescriptors = [ eventDescriptors ];
+	}
+
+	eventDescriptors.forEach( ( event ) => {
+		const { type, name } = event;
+
+		switch ( type ) {
+			case 'command':
+				registerCommandListener( name, event.state, callback );
+				break;
+
+			case 'window-event':
+				registerWindowEventListener( name, callback );
+				break;
+		}
+	} );
+}
+
+export function flushListeners() {
+	abortController.abort();
+	callbacksByEvent.clear();
+
+	abortController = new AbortController();
+}
+
+function registerCommandListener(
+	command: CommandEventDescriptor['name'],
+	state: CommandEventDescriptor['state'],
+	callback: ListenerCallback
+) {
+	registerWindowEventListener( `elementor/commands/run/${ state }`, ( e ) => {
+		const shouldRunCallback = e.type === 'command' && e.command === command;
```

Comment:
why both conditions?
+
Not sure that the variable is needed..

Reply:
both conditions are to satisfy TS, the variable is for readability (@Nevoss's idea)